engine/src/main/java/org/terasology/rendering/nui/layouts/SingleElementLayout.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.layouts;
+
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.rendering.nui.*;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ *
+ */
+public class SingleElementLayout extends CoreLayout<LayoutHint> {
```

Comment:
Since the purpose of this widget isn't layouting I would choose a name that fits more it's purpose.

e.g. some idears for names: PreviewWidget, ElementViewerWidget, ContainmentWidget

Also of course I would then let the widget no longer extend CoreLayout but AbstractWidget. Also I would then make it not have a addWidget method but a setWidget/setContainedWidget method. Having a addWidget method that implictly removes the previous element is a bit of strange.

If you really want to keep this name, I am also fine with that but then please add a javadoc that explains it's purpose.